examples/src/agents/requests.ts

```diff
@@ -0,0 +1,107 @@
+import { OpenAI, LLMChain } from "langchain";
+import { ZeroShotAgent, AgentExecutor } from "langchain/agents";
+import * as fs from "fs";
+import * as yaml from "js-yaml";
+import {
+  JsonSpec,
+  JsonToolkit,
+  RequestsToolkit,
+  DynamicTool,
+} from "langchain/tools";
+
+export const run = async () => {
+  // create the json agent
+  let data: object;
+  try {
+    const yamlFile = fs.readFileSync("openai_openapi.yaml", "utf8");
+    data = yaml.load(yamlFile) as object;
+    if (!data) {
+      console.error("Could not load YAML file");
+    }
+  } catch (e) {
+    console.error(e);
+    return;
+  }
+
+  const toolkit = new JsonToolkit(new JsonSpec(data));
+  const model = new OpenAI({ temperature: 0 });
+
+  const jsonAgent = ZeroShotAgent.asJsonAgent(model, toolkit);
+  const jsonExecutor = AgentExecutor.fromAgentAndTools({
```

Comment:
I wonder if we should have a method on agentexecutor to turn it into a (dynamic)tool. Would make it slightly easier to do so, and remind people that it is possible to do

Reply:
yeah good point, can add in a followup PR